table.go

```diff
@@ -87,12 +87,29 @@ func (t *TableMap) SetUniqueTogether(fieldNames ...string) *TableMap {
 			"gorp: SetUniqueTogether: must provide at least two fieldNames to set uniqueness constraint."))
 	}
 
-	columns := make([]string, 0)
+	columns := make([]string, 0, len(fieldNames))
 	for _, name := range fieldNames {
 		columns = append(columns, name)
 	}
-	t.uniqueTogether = append(t.uniqueTogether, columns)
-	t.ResetSql()
+
+	alreadyExists := false
+checkDuplicates:
```

Comment:
I'm not really comfortable merging named loops if they can be avoided.  I'll refactor this next time I get a chance (hopefully tomorrow) and send you a PR for your PR.